src/txmempool.h

```diff
@@ -215,7 +215,7 @@ class CTxMemPool
     }
 
     // Lookup for the transaction with the specific hash (txid).
-    virtual bool lookup(const uint256 &txid, CTransaction& tx) const;
+    virtual bool lookup(const uint256 &txid, CTransaction& tx, uint32_t &nBlockHeight) const;
```

Comment:
I think you can even change uint32_t &nBlockHeight parameter to uint32_t *pnBlockHeight.
and again: if (pnBlockHeight) -> you have to make sure it is assigned in ALL paths inside CMemPool::lookup.
If you make that parameter optional -> many places you've changed will also go away.